### ISSUE TITLE
Support Soft Hyphens

### ISSUE THREAD

**Issue by dsuppiger** [renehamburger/blinx.js#28]
If the book name of a bible reference contains soft hyphens (`&shy;` or `&#173;`) it is not recognized.

Example (HTML):
`Ze&shy;phan&shy;iah 3:17`

**Comment by renehamburger** [comment]
Thanks for pointing this out! I'll try to address this shortly.

**Comment by renehamburger** [comment]
@dsuppiger, please try the most recent version 0.3.11. Soft hyphens are now removed before the references are parsed.

This fix should really be applied downstream in the parser itself, but the project isn't maintained and it's so complex that I'm hesitant to make such a (even small) change there right now.

**Comment by dsuppiger** [comment]
Thanks, but unfortunately it doens't quite work.

The more soft hyphens (min. 2) precede a reference within the same tag, the more the anchors are displaced to the right. The references are recognized correctly though.

**Comment by renehamburger** [comment]
Have you got a public website I could check this at? Which browser are you using?

**Comment by dsuppiger** [comment]
I had the issue on [this](http://www.josiablog.de/2018/10/teil-i-vergebung-von-anderen-menschen-in-anspruch-nehmen) page, but now removed the soft hyphens.
It can be reproduced with a simple HTML script like this though:

```html
<html>
<head>
    <title>Blinx.js Test</title>
</head>
<body>
    <p>Ge&shy;ne&shy;sis 1:1, Ge&shy;ne&shy;sis 2:1, Ge&shy;ne&shy;sis 3:1</p>

    <script src="https://cdn.rawgit.com/renehamburger/blinx.js/v0.3.11/dist/blinx.js" defer data-blinx="{ language: 'en' }"></script>
</body>
</html>
```

I tested it with Google Chrome.

**Comment by renehamburger** [comment]
Thanks for providing the example, @dsuppiger. I hadn't tested multiple references with soft-hyphens before. This is fixed in [v0.3.12](https://github.com/renehamburger/blinx.js/releases/tag/v0.3.12).
